src/widgets/scrollbar.rs

```diff
@@ -0,0 +1,698 @@
+
+    fn render(self, area: Rect, buf: &mut Buffer, state: &mut Self::State) {
+        // info!("-----------------------------");
+        // info!("{:?}", area);
+
+        let is_vertical = match self.orientation {
+            ScrollbarOrientation::Vertical => true,
+            ScrollbarOrientation::Horizontal => false,
+        };
+
+        // Decrease scrollbar area if a begin arrow is present
+        let area = if self.begin_symbol.is_some() {
+            if is_vertical {
+                // For vertical scrollbar, reduce the height by one
+                Rect::new(
+                    area.x,
+                    area.y + 1,
+                    area.width,
+                    area.height.saturating_sub(1),
+                )
+            } else {
+                // For horizontal scrollbar, reduce the width by one
+                Rect::new(
+                    area.x + 1,
+                    area.y,
+                    area.width.saturating_sub(1),
+                    area.height,
+                )
+            }
+        } else {
+            area
+        };
+        // Further decrease scrollbar area if an end arrow is present
+        let area = if self.end_symbol.is_some() {
+            if is_vertical {
+                Rect::new(area.x, area.y, area.width, area.height.saturating_sub(1))
+            } else {
+                Rect::new(area.x, area.y, area.width.saturating_sub(1), area.height)
+            }
+        } else {
+            area
+        };
+
+        // info!("{:?}", area);
+
+        // Define main size, cross position, and main position based on scrollbar orientation
+        let (main_size, main_position, cross_position, other_cross_position) =
+            if is_vertical && self.mirror {
+                (area.height, area.top(), area.left(), area.right())
+            } else if is_vertical {
+                (
+                    area.height,
+                    area.top(),
+                    area.right().saturating_sub(1),
+                    area.left(),
+                )
+            } else if self.mirror {
+                (area.width, area.left(), area.top(), area.bottom())
+            } else {
+                (
+                    area.width,
+                    area.left(),
+                    area.bottom().saturating_sub(1),
+                    area.top(),
+                )
+            };
+
+        // Bail on rendering if the main dimension (vertical or horizontal)
+        // doesn't have room to draw, or if length of the contents to be scrolled is zero.
+        if main_size == 0 || state.content_length == 0 {
+            return;
+        }
+
+        // Bail on rendering if the cross dimension (horizontal or vertical)
+        // doesn't have room to draw.
+        if (self.mirror && cross_position >= other_cross_position)
+            || (!(self.mirror) && cross_position <= other_cross_position)
+        {
+            return;
+        }
```

Comment:
I previously chose to not render the scrollbar vertically if the width == 1 but I think that is not needed. I removed it now and it will render a vertical scrollbar even if the width of the rect is 1.